src/common/checksum.c

```diff
@@ -345,10 +429,10 @@ daos_csummer_set_dcbs(struct daos_csummer *obj, daos_csum_buf_t *dcbs,
 		D_ASSERT(recx_idx < iod->iod_nr);
 		recx = &iod->iod_recxs[recx_idx];
 
-		csum_nr = daos_recx_calc_chunks(*recx,
-						iod->iod_size,
+		if (!csum_iod_is_supported(obj->dcs_chunk_size, iod))
```

Comment:
can this check be outside the loop?

Reply:
no, because the iods are enumerated in this function as well.